README.md

```diff
@@ -1598,10 +1598,1953 @@ end
+<br>
+
+Let's add an onChange event to our select input that will call the updateFilter function. In order to accomplish that we must bind the object 'this' to our function call.
+
+_filter_selects.js.jsx
+```
+render (){
+    return(
+      <div>
+        <label>
+          Filter By
+        </label>
+        <select className="form-control" onChange={this.updateFilter.bind(this)}>
+        
+        ...
+```
+<br>
+Now we need to create an updateFilter function within the FilterSelect component.
+
+_filter_selects.js.jsx
+```
+updateFilter(event) {
+    this.props.filter(event.target.value)
+  }
+```
+Since we bound the object 'this' to the function call in our updateFilter, 'this' refers to the entire component. This gives us access to all of the properties of the FilterSelect component. We need the filter property, which we pass into the component when we render it in the Body. The filter property points to the filterSkills function in the Body component. Let's pass in the value of the select input, which we get from the event object.
+```
+event.target.value
+```
+
+If everything is working properly, when you reload the page and select an option from the selector, you should hit the debugger within the filterSkills function. You should also see "about to filter" logged to the console. From the console, you can type "value" and you will get the value you selected from the dropdown.
+<br>
+Next we need to actually make the filterSkills function filter our skills list based on the value passed in from the select input.
+
+_body.js.jsx
+```
+filterSkills(value) {
+    // Create a variable called skills set to an empty string
+    var skills = ''
+    // check if there is an existing allSkills property in the state object
+    if(this.state.allSkills) {
+      // if allSkills exists, we know that the skill list has been filtered once
+      // before, so we set the skills variable to equal all the skills.
+      skills = this.state.allSkills;
+    } else {
+      // if this is the first time filtering, we set the variable skills to equal
+      // all of the skills.
+      skills = this.state.skills;
+    }
+    // Filter out any skills that do not match the selected value.
+    skills = skills.filter((skill) => { return skill.level === value });
+    // If this.state.allSkills exists
+    if(this.state.allSkills) {
+      // We can just set the state of the skills to be our skills variable
+      this.setState({ skills: skills });
+    } else {
+      // We additionally set allSkills to be the original list of skills.
+      this.setState({ skills: skills, allSkills: this.state.skills });
+    }
+  }
+```
+The above code will filter on 'bad', 'halfbad', and 'fantastic' skill levels, but has two big problems. First, the 'All' option will not return all of the skills. Seco# Creact
+## Crud + React
```

Comment:
It seems like this is still happening, the `sections` piece is below, and then you have another copy of section 12 after that. 

The diff is a bit weird, now it looks like you are adding the entire tutorial. To get a cleaner diff, I would remove everything you have above, and update the section below.